src/main/java/org/cancogenvirusseq/muse/components/PayloadFileMapper.java

```diff
@@ -153,7 +158,11 @@ private static ObjectNode convertRecordToPayload(
           } else if (value.toString().trim().equals("")) {
             // empty string map to null value
             return "null";
-          } else {
+          } else if(value.toString().split(multiTagDelimiter).length>1) {
+            //for multiple tags in single cell
+            return StringUtils.stringToArrayOfStrings(value.toString(), multiTagDelimiter);
+          }
```

Comment:
I think we need to update the if condition here... we only want to convert to array for the specified columns (`key == "purpose_of_sequencing"`) and we always want to convert to array for those columns, even for length of 0 or 1.

Can we change this to only apply the `stringToArrayOfStrings` for the columns requested?

Reply:
The code now maintains a list of columnKeys